src/server/MCFadapter.js

```diff
@@ -128,34 +128,35 @@ export class MCFAdapter {
       })
   }
 
-  heatMap = genes => {
-    // Convert genes array to genes array for psql
-    let genesList = _.join(genes, ',')
-    genesList = `'{` + genesList + `}'`
-
-    return this.pool.query(`
-      SELECT
-        key,
-        gene,
-        mcf10a_fpkm,
-        mcf7_fpkm,
-        mcf10a_log2,
-        mcf7_log2
-      FROM
-        mcf10a_vs_mcf7
-      JOIN
-        unnest(${genesList}::varchar[]) WITH ORDINALITY t(gene, key) USING (gene)
-      ORDER BY t.key
-    `)
-  }
+  heatMap = (genes, type) => {
+    if (type === 'fpkm') {
+      // Convert genes array to genes array for psql
+      let genesList = _.join(genes, ',')
+      genesList = `'{` + genesList + `}'`
 
-  intronAnalysisHeatmap = gene => {
+      return this.pool.query(`
+        SELECT
+          key,
+          gene,
+          mcf10a_fpkm,
+          mcf7_fpkm,
+          mcf10a_log2,
+          mcf7_log2
+        FROM
+          mcf10a_vs_mcf7
+        JOIN
+          unnest(${genesList}::varchar[]) WITH ORDINALITY t(gene, key) USING (gene)
+        ORDER BY t.key
+      `)
+    }
+    console.log(genes, type)
+    // type === 'psi'
     return this.pool
       .query(
         `
       SELECT intron_number,
       mcf10a_log2_psi, mcf7_log2_psi
-      FROM mcf_intron_psi WHERE gene = '${gene}' ORDER BY intron_number
+      FROM mcf_intron_psi WHERE gene = '${genes.toUpperCase()}' ORDER BY intron_number
```

Comment:
is it okay to call functions withing a query like this?

Reply:
Can we keep the `intronAnalysisHeatmap` function? This change is going to conflict with @dordep's mongodb conversion.

Just have it call the correct heatmap function in the handler for `app.post('/api/heatmap')`. 

Later once we merge the mongodb PR, you can consolidate them under one heatmap function.